probe/ebpf/flow.c

```diff
@@ -347,6 +356,28 @@ int bpf_flow_table(struct __sk_buff *skb)
 		update_metrics(skb, prev, tm,
 			flow.link_layer._hash_src == prev->link_layer._hash_src);
 		__sync_fetch_and_add(&prev->last, tm - prev->last);
+
+		if(prev->transport_layer.port_src==flow.transport_layer.port_src){
+			if(prev->transport_layer.ab_syn==0){
+				__sync_fetch_and_add(&prev->transport_layer.ab_syn, flow.transport_layer.ab_syn);
+			}
+			if(prev->transport_layer.ab_fin==0){
+				__sync_fetch_and_add(&prev->transport_layer.ab_fin, flow.transport_layer.ab_fin);
+			}
+			if(prev->transport_layer.ab_rst==0){
+				__sync_fetch_and_add(&prev->transport_layer.ab_rst, flow.transport_layer.ab_rst);
+			}
+		} else{
+			if(prev->transport_layer.ba_syn==0){
+				__sync_fetch_and_add(&prev->transport_layer.ba_syn, flow.transport_layer.ab_syn);
```

Comment:
flow.transport_layer.ba_*

Reply:
"flow" has only now been generated because of the current packet that triggered this code being passed from flow.a to flow.b and therefore flow.transport_layer.ba*=0 always. what we do here is in case prev.src=flow.dst and prev.dst=flow.src (meaning thath this packet is a response from the original destination to the original soucre) update the flags accordingly (in the right direction).